sympy/core/relational.py

```diff
@@ -1138,3 +1033,308 @@ def _eval_relation(cls, lhs, rhs):
     '<': StrictLessThan,
     'lt': StrictLessThan,
 }
+
+
+def _n2(a, b):
+    """Return (a - b).evalf(2) if a and b are comparable, else None.
+    This should only be used when a and b are already sympified.
+    """
+    # /!\ it is very important (see issue 8245) not to
+    # use a re-evaluated number in the calculation of dif
+    if a.is_comparable and b.is_comparable:
+        dif = (a - b).evalf(2)
+        if dif.is_comparable:
+            return dif
+
+
+def fuzzy_resolve(lhs, rhs, op, cls):
+    """
+    Resolves op to S.true if cmp returns true.
+    or to S.false if cmp returns false
+    or to an unevaluated if cmp returns none.
+    """
+    val = op(lhs, rhs)
+    if val is None:
+        return cls(lhs, rhs, evaluate=False)
+    elif isinstance(val,bool):
+        if val is True:
+            return S.true
+        else:
+            return S.false
+    else:
+        return val
+
+
+def fuzzy_not(val):
+    if val is None:
+        return val
+    else:
+        return not val
+
+
+
+def eval_dispatch(lhs, rhs, op):
+    """
+    Dipatches to the appropriate equality operator on Expr/Basic
+    objects.
+    """
+    # if one of the either the lhs, or the rhs has an _eval_is_X method
+    # return the result, otherwise do nothing
+    retval = None
+    eval_str = "_eval_"
+    negate_op = False
+    if op == "<":
+        eval_str = eval_str + "is_ge"
+        negate_op = True
```

Comment:
The "<" and ">=" operations are only mutually exclusive when both operands are guaranteed to be extended real. Is that checked somewhere else?

Reply:
no, it is not. I'll fix that.